controllers/kmsissuer_controller.go

```diff
@@ -18,24 +18,30 @@ package controllers
 
 import (
 	"context"
-	"errors"
-	"fmt"
-
+	"crypto/x509"
 	"crypto/x509/pkix"
 	"encoding/pem"
+	"errors"
+	"fmt"
 	"time"
 
 	kmsiapi "github.com/Skyscanner/kms-issuer/api/v1alpha1"
 	"github.com/Skyscanner/kms-issuer/pkg/kmsca"
 	"github.com/go-logr/logr"
 	core "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/client-go/tools/record"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/manager"
 )
 
+const (
+	// DefaultCertDuration is the default duration the CA certificate is valid for.
+	DefaultCertDuration = time.Hour * 24 * 365 * 3 // 3 years
+)
+
 // KMSIssuerReconciler reconciles a KMSIssuer object.
 type KMSIssuerReconciler struct {
```

Comment:
I've noticed that `Log` is an exported/public field. Do we really need it to be one?

Reply:
This part of the code is generated by `kubebuilder`. It is also used to instantiate a custom Controller in the tests.
https://github.com/Skyscanner/kms-issuer/blob/main/controllers/suite_test.go#L91-L96